### DISCUSSION TITLE
Event Handler not triggering wait task

### DISCUSSION THREAD

**Discussion by kathir0301** [Netflix/conductor#3208] [Q&A] [answered]
I am trying to run the events as mentioned in the conductor documentation but the wait task is not getting triggered remains inprogress state . Here is my workflow and event handler definiton.

**Workflow 1**
{
  "name": "workflow_start",
  "description": "A test workflow to start another workflow with EventHandler",
  "version": 1,
  "tasks": [
    {
      "name": "workflow_one",
      "taskReferenceName": "workflow_one_event",
      "type": "EVENT",
      "sink": "conductor"
    },
    {
      "name": "complete_tasks_final",
      "taskReferenceName": "complete_tasks_final",
      "type": "WAIT"
    }
  ]
}

**Workflow2**

{
  "name": "workflow_event",
  "description": "A test workflow which is started by EventHandler, and then goes on to complete task in another workflow.",
  "version": 1,
  "tasks": [
    {
      "name": "workflow_two",
      "taskReferenceName": "workflow_two_event",
      "inputParameters": {
        "sourceWorkflowId": "${workflow.input.sourceWorkflowId}"
      },
      "type": "EVENT",
      "sink": "conductor"
    }
  ]
}

**Event Handler one:**
{
  "name": "test_start_workflow",
  "event": "conductor:workflow_start:workflow_one_event",
  "actions": [
    {
      "action": "start_workflow",
      "start_workflow": {
        "name": "workflow_event",
        "input": {
          "sourceWorkflowId": "${workflowInstanceId}"
        }
      }
    }
  ],
  "active": true
}

Event handler 2:

{
  "name": "workflow_two",
  "event": "conductor:workflow_event:workflow_two_event",
  "actions": [
    {
        "action": "complete_task",
        "complete_task": {
            "workflowId": "${sourceWorkflowId}",
            "taskRefName": "complete_tasks_final"
         }
    }
  ],
  "active": true
}

My wait task not gets completed and stays in progress. 

Any help is appreciated

**Comment by c4lm** [comment, marked as answer]
From what I see everything is defined properly. I have just tested it and my wait task did get completed. Have you tried waiting for a few seconds (it is not immediate)?
Definitions:
```
{
  "createTime": 1661890360126,
  "name": "workflow_start",
  "description": "A test workflow to start another workflow with EventHandler",
  "version": 1,
  "tasks": [
    {
      "name": "workflow_one",
      "taskReferenceName": "workflow_one_event",
      "inputParameters": {},
      "type": "EVENT",
      "startDelay": 0,
      "sink": "conductor",
      "optional": false,
      "asyncComplete": false
    },
    {
      "name": "complete_tasks_final",
      "taskReferenceName": "complete_tasks_final",
      "inputParameters": {},
      "type": "WAIT",
      "startDelay": 0,
      "optional": false,
      "asyncComplete": false
    }
  ],
  "inputParameters": [],
  "outputParameters": {},
  "schemaVersion": 2,
  "restartable": true,
  "workflowStatusListenerEnabled": false,
  "ownerEmail": "example@email.com",
  "timeoutPolicy": "ALERT_ONLY",
  "timeoutSeconds": 0,
  "variables": {},
  "inputTemplate": {}
}
```
```
{
  "createTime": 1661890427788,
  "name": "workflow_event",
  "description": "A test workflow which is started by EventHandler, and then goes on to complete task in another workflow.",
  "version": 1,
  "tasks": [
    {
      "name": "workflow_two",
      "taskReferenceName": "workflow_two_event",
      "inputParameters": {
        "sourceWorkflowId": "${workflow.input.sourceWorkflowId}"
      },
      "type": "EVENT",
      "startDelay": 0,
      "sink": "conductor",
      "optional": false,
      "asyncComplete": false
    }
  ],
  "inputParameters": [],
  "outputParameters": {},
  "schemaVersion": 2,
  "restartable": true,
  "workflowStatusListenerEnabled": false,
  "ownerEmail": "example@email.com",
  "timeoutPolicy": "ALERT_ONLY",
  "timeoutSeconds": 0,
  "variables": {},
  "inputTemplate": {}
}
```
```
{
  "name": "test_start_workflow",
  "event": "conductor:workflow_start:workflow_one_event",
  "actions": [
    {
      "action": "start_workflow",
      "start_workflow": {
        "name": "workflow_event",
        "input": {
          "sourceWorkflowId": "${workflowInstanceId}"
        }
      },
      "expandInlineJSON": false
    }
  ],
  "active": true
}
```
```
{
  "name": "workflow_two",
  "event": "conductor:workflow_event:workflow_two_event",
  "actions": [
    {
      "action": "complete_task",
      "complete_task": {
        "workflowId": "${sourceWorkflowId}",
        "taskRefName": "complete_tasks_final",
        "output": {}
      },
      "expandInlineJSON": false
    }
  ],
  "active": true
}
```
<img width="1354" alt="image" src="https://user-images.githubusercontent.com/13348474/187535356-fea954a0-db24-461d-94c6-63da43431af0.png">

**Reply by kathir0301** [reply]
Thanks for testing the workflow . I tried running with default settings and complete tasks is not triggered .

Any idea how to troubleshoot this?

Appreciate your respose

**Reply by c4lm** [reply]
The correct way to troubleshoot this is to set up debug breakpoints:
in com.netflix.conductor.core.execution.tasks.Event::execute and ::start
in com.netflix.conductor.core.events.SimpleActionProcessor::execute
And then walking step by step.

Do lines like these appear in the logs?
```
com.netflix.conductor.redis.dao.RedisExecutionDAO [] - updating event execution test.EVENT_EXECUTION.test_start_workflow.conductor:workflow_start:workflow_one_event.a2edc540-5f78-487d-a1fb-ea5643cd1d77

com.netflix.conductor.redis.dao.RedisExecutionDAO [] - updating event execution test.EVENT_EXECUTION.workflow_two.conductor:workflow_event:workflow_two_event.1459c382-2667-4a81-8109-8a777eb8aee1
```

There should be two lines like these, first for handling the workflow start event, some time later second one for the complete task event.
If there are two such lines, but the task was not completed, double check the definitions to make sure there are no typos, make sure both event handlers are present and active (by querying `/api/event`)
If you use the definitions posted above with the latest version of Conductor, they will work as expected.

**Reply by kathir0301** [reply]
@c4lm Thank you for response , really appreciate it
